xarray/backends/netCDF4_.py

```diff
@@ -473,9 +463,9 @@ def prepare_variable(
                 "netCDF4 does not yet support setting a fill value for "
                 "variable-length strings "
                 "(https://github.com/Unidata/netcdf4-python/issues/730). "
-                "Either remove '_FillValue' from encoding on variable %r "
+                f"Either remove '_FillValue' from encoding on variable {name!r} "
                 "or set {'dtype': 'S1'} in encoding to use the fixed width "
-                "NC_CHAR type." % name
+                "NC_CHAR type."
             )
```

Comment:
@shoyer, unless I am missing something, this issue has been resolved upstream (https://github.com/Unidata/netcdf4-python/issues/730), right?

Reply:
Yes, it would be good to adjust the corresponding test and remove this

Reply:
Okay.. Will address this in a separate PR.